catkit/hardware/boston/DmCommand.py

```diff
@@ -182,22 +182,33 @@ def load_dm_command(path, dm_num=1, flat_map=False, bias=False, as_volts=False):
 
 
 def get_flat_map_volts(dm_num):
+    """
+    Get the flat map for a given dm. The flat map is in volts for each actuator.
+    :param dm_num: Which DM to lead the command for.
+    :return: flat map for the selected DM. This function caches the map to avoid multiple disk access.
+    """
     calibration_data_package = CONFIG_INI.get("optics_lab", "calibration_data_package")
     calibration_data_path = os.path.join(catkit.util.find_package_location(calibration_data_package),
                                          "hardware",
                                          "boston")
 
     global flat_map_dm1, flat_map_dm2
 
-    if flat_map_dm1 is None:
-        fname = CONFIG_INI.get("boston_kilo952", "flat_map_dm1")
-        flat_map_dm1 = fits.getdata(os.path.join(calibration_data_path, fname))
+    if dm_num == 1:
+        if flat_map_dm1 is None:
+            fname = CONFIG_INI.get("boston_kilo952", "flat_map_dm1")
+            flat_map_dm1 = fits.getdata(os.path.join(calibration_data_path, fname))
+
+        return flat_map_dm1
 
-    if flat_map_dm2 is None:
-        fname = CONFIG_INI.get("boston_kilo952", "flat_map_dm2")
-        flat_map_dm2 = fits.getdata(os.path.join(calibration_data_path, fname))
+    if dm_num == 2:
+        if flat_map_dm2 is None:
+            fname = CONFIG_INI.get("boston_kilo952", "flat_map_dm2")
+            flat_map_dm2 = fits.getdata(os.path.join(calibration_data_path, fname))
 
-    return flat_map_dm1 if dm_num == 1 else flat_map_dm2
+        return flat_map_dm2
+
+    raise ValueError('dm_num should be either 1 or 2.')
```

Comment:
Not gonna ask you to redo this, but this sort of logic pattern is neater and more resilient to future changes when self-contained as an ``else`` to an ``elif``, i.e., 
```python
if dm_num == 1:
  ...
elif dm_num == 2:
  ...
else:
   raise
```

Reply:
I considered doing a dictionary approach too. That would have been even cleaner. PEP 634 is gonna be so good for this type of stuff too. I've been waiting for that for a long time.